pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey = tag.MustNewKey("elapsed")
+
+	reminderTypeKey = tag.MustNewKey("reminder_type")
+
+	executionTypeKey = tag.MustNewKey("execution_type")
+)
+
+const (
+	StatusSuccess   = "success"
+	StatusFailed    = "failed"
+	StatusRetryable = "retryable"
+	CreateWorkflow  = "create_workflow"
+	GetWorkflow     = "get_workflow"
+	AddEvent        = "add_event"
+	PurgeWorkflow   = "purge_workflow"
+	Activity        = "activity"
+	Workflow        = "workflow"
+	WorkflowEvent   = "event"
+	Timer           = "timer"
+)
+
+type workflowMetrics struct {
+	// Total Successful and Failed workflow operation requests
+	workflowOperationsCount *stats.Int64Measure
+	// Workflow operation request's Response latency
+	workflowOperationsLatency *stats.Float64Measure
+	// Total workflow/activity reminders created
+	workflowRemindersCount *stats.Int64Measure
+	// Total successful/failed workflow/activity executions
+	workflowExecutionCount *stats.Int64Measure
+	// Time taken to run a workflow to completion
+	workflowExecutionLatency *stats.Float64Measure
+	// Latency between execution request and actual execution
+	workflowSchedulingLatency *stats.Float64Measure
+	appID                     string
+	enabled                   bool
+}
+
+func newWorkflowMetrics() *workflowMetrics {
+	return &workflowMetrics{
+		workflowOperationsCount: stats.Int64(
+			"runtime/workflow/operations/total",
+			"The number of successful/failed workflow operation requests.",
+			stats.UnitDimensionless),
+
+		workflowOperationsLatency: stats.Float64(
+			"runtime/workflow/operations/latency",
+			"The latencies of responses for workflow operation requests.",
+			stats.UnitMilliseconds),
+		workflowRemindersCount: stats.Int64(
+			"runtime/workflows/reminders/total",
```

Comment:
Should we change "workflows" here and below to "workflow" (no 's') to maintain consistency with the other elements of the struct?

Reply:
Agree.

Reply:
fixed